third_party/java/arcs/build_defs/internal/plan.bzl

```diff
@@ -0,0 +1,85 @@
+"""Arcs Plan Generation Rules"""
+
+load(
+    "//third_party/java/arcs/build_defs/internal:kotlin.bzl",
+    "ARCS_SDK_DEPS",
+    "arcs_kt_library",
+)
+
+# Note: Once this is mature, it will replace arcs_kt_plan
+def arcs_plan_generation(name, package, srcs = [], deps = [], visibility = None):
```

Comment:
It seems like most bazel rules are a noun that defines the type of target. The "generation" on this makes it seem like the rule will define a generation of arcs plans.   `arcs_kt_plan` seemed more appropriate of a name.  Maybe for now it could be `arcs_kt_plan_2`, and when it's mature/ready we can just remove the `_2`.

Reply:
Sure, I'm happy with that. I'm not attached to any particular name for this macro.